plugins/cachekey/configs.cc

```diff
@@ -60,26 +60,67 @@ ConfigElements::setInclude(const char *arg)
 }
 
 static void
-setPattern(MultiPattern &multiPattern, const char *arg)
+setPattern(MultiPattern *multiPattern, const char *arg)
```

Comment:
OK, makes sense.

Reply:
One of the new callers gets `multiPattern` from a value in a `std::map`. We cannot store references as values in a `map`, so the signature here was changed to use pointers instead. We might want to add the null-check just to be safe against future changes to the code.

Reply:
@d2r I believe we can go either way, that is why I did not think it is worth the time arguing.

Initially I wrote it as a reference for the exact reasons mentioned by @a-canary.

But you are right too, the new caller gets a pointer value from the map and this is why I changed to to a pointer. Well, it might be also because I don't like dereferencing a pointer when passing by reference much (may be because I learned C before C++ :blush:) but the same way I also don't care for certain colors that others like a lot.

The main point is to make sure we don't dereference `nullptr`. It can be checked in the caller, in the called function or both. Two callers will never dereference `nullptr` by nature, for the one who "can try to do it" I made sure it doesn't (otherwise I have a bug)

Reply:
Agree. I was clarifying for @a-canary why it is the signature changed from reference to pointer. I think we both suggested a null-check. If instead we want the caller to ensure `multiPattern` is not null, then we can just add a small code comment saying `multiPattern is assumed to be a non-null value` or something like that.